pkg/catalog/types.go

```diff
@@ -25,10 +24,6 @@ type MeshCatalog struct {
 	compute.Interface
```

Comment:
It's not a part of this PR (and it seems like this PR largely avoids this pitfall by explicitly calling out `MeshCatalog.Interface`), but embedding this interface seems to imply a tighter coupling between MeshCatalog and a `compute.Interface` than desired IMO. Interface embedding is already discouraged/prohibited in Uber's [style guide](https://github.com/uber-go/guide/blob/master/style.md#avoid-embedding-types-in-public-structs), and it's just confusing here. In my mind, a `MeshCatalog` has a dependency on a `compute.Interface` it is not itself a `compute.Interface` like embedding may imply. /cc @steeling for some possible context

Reply:
Personally, I disagree with Uber's conclusion on embedding interface. Although it might be something to take a harder look at, I think it has its places.

For us here, I think there's a good chance the meshcatalog and compute.Client eventually get merged. Embedding the interface allows abstracting the methods in compute.Interface, such that the response.go files don't need to know which layer of the domain they are accessing.

In fact, I recommend we remove `meshCatalog.Interface.<MethodCall>`, and directly call the method from meshCatalog. As we deconstruct some of the meshcatalog calls it will make for less changes.

* This gets heavy into opinions, so only providing mine as a reference point :)

Reply:
https://github.com/openservicemesh/osm/issues/4863 has the full description